source/common/upstream/subset_lb.cc

```diff
@@ -25,19 +26,35 @@ SubsetLoadBalancer::SubsetLoadBalancer(
     LoadBalancerType lb_type, PrioritySet& priority_set, const PrioritySet* local_priority_set,
     ClusterLbStats& stats, Stats::Scope& scope, Runtime::Loader& runtime,
     Random::RandomGenerator& random, const LoadBalancerSubsetInfo& subsets,
-    const absl::optional<envoy::config::cluster::v3::Cluster::RingHashLbConfig>&
-        lb_ring_hash_config,
-    const absl::optional<envoy::config::cluster::v3::Cluster::MaglevLbConfig>& lb_maglev_config,
-    const absl::optional<envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>&
-        round_robin_config,
-    const absl::optional<envoy::config::cluster::v3::Cluster::LeastRequestLbConfig>&
-        least_request_config,
+    OptRef<const envoy::config::cluster::v3::Cluster::RingHashLbConfig> lb_ring_hash_config,
+    OptRef<const envoy::config::cluster::v3::Cluster::MaglevLbConfig> lb_maglev_config,
+    OptRef<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig> round_robin_config,
+    OptRef<const envoy::config::cluster::v3::Cluster::LeastRequestLbConfig> least_request_config,
     const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config,
     TimeSource& time_source)
-    : lb_type_(lb_type), lb_ring_hash_config_(lb_ring_hash_config),
-      lb_maglev_config_(lb_maglev_config), round_robin_config_(round_robin_config),
-      least_request_config_(least_request_config), common_config_(common_config), stats_(stats),
-      scope_(scope), runtime_(runtime), random_(random), fallback_policy_(subsets.fallbackPolicy()),
+    : lb_type_(lb_type),
+      lb_ring_hash_config_(
+          lb_ring_hash_config.has_value()
+              ? std::make_unique<const envoy::config::cluster::v3::Cluster::RingHashLbConfig>(
```

Comment:
nit: Can you use `ref()` here instead of `value()` to be more explicit that the copy constructor is invoked.

Reply:
Done